native/dll/Win32DLL/WindowsServiceControl.cpp

```diff
@@ -651,47 +647,37 @@ GetAccountSid(
     // 
     // initial memory allocations
     // 
-    if((*Sid=HeapAlloc(
-                    GetProcessHeap(),
-                    0,
-                    cbSid
-                    )) == NULL) __leave;
-
-    if((ReferencedDomain=(LPTSTR)HeapAlloc(
-                    GetProcessHeap(),
-                    0,
-                    cchReferencedDomain * sizeof(TCHAR)
-                    )) == NULL) __leave;
+    if((*Sid=HeapAlloc(GetProcessHeap(), 0, cbSid)) == nullptr) __leave;
+
+    if((ReferencedDomain=(LPTSTR)HeapAlloc(GetProcessHeap(), 0, cchReferencedDomain * sizeof(TCHAR))) == NULL) __leave;
 
     // 
     // Obtain the SID of the specified account on the specified system.
     // 
-    while(!LookupAccountName(
-                    SystemName,         // machine to lookup account on
-                    AccountName,        // account to lookup
-                    *Sid,               // SID of interest
-                    &cbSid,             // size of SID
-                    ReferencedDomain,   // domain account was found on
-                    &cchReferencedDomain,
-                    &peUse
-                    )) {
-        if (GetLastError() == ERROR_INSUFFICIENT_BUFFER) {
-            // 
-            // reallocate memory
-            // 
-            if((*Sid=HeapReAlloc(
-                        GetProcessHeap(),
-                        0,
-                        *Sid,
-                        cbSid
-                        )) == NULL) __leave;
+	if ((Sid != nullptr) && (ReferencedDomain != NULL))

```

Comment:
You don't need to check Sid or ReferencedDomain because they are already being checked above and the __leave is done to terminate the section.

Reply:
Done